packages/core/src/browser/saveable.ts

```diff
@@ -87,10 +87,16 @@ export namespace Saveable {
         return !!getDirty(arg);
     }
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    export async function save(arg: any): Promise<void> {
+    export async function save(arg: any, options?: SaveOptions): Promise<void> {
         const saveable = get(arg);
+        /**
+         * skipFormatting is true if:
+         * - The function receives skipFormattingOnSave flag from 'options' object (ie: from command `saveWithoutFormatting`).
+         * - The file is not dirty and performing save in multiple files (saveAll command).
+         */
+        const skipFormatting = options?.skipFormattingOnSave || (!saveable?.dirty && options?.savingMultiple);
```

Comment:
This check can be performed in the `if (saveable)` block:

```ts
const skipFormatting = options?.skipFormattingOnSave || (!saveable.dirty && options?.savingMultiple);
```

It can save us computation when `saveable` is `undefined`, as `skipFormatting` is only used when actually saving.